hydra/core/callbacks.py

```diff
@@ -0,0 +1,51 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import warnings
+from typing import Any
+
+from omegaconf import DictConfig
+
+from hydra.core.utils import JobReturn
+from hydra.errors import HydraException
+from hydra.utils import instantiate
+
+
+class Callbacks:
+    def __init__(self, config: DictConfig) -> None:
+        self.callbacks = []
+        if config.hydra.callbacks is None:
+            raise HydraException("Hydra Callbacks is not configured")
+
+        for params in config.hydra.callbacks.values():
+            self.callbacks.append(instantiate(params))
+
+    def _notify(self, function_name: str, **kwargs: Any) -> None:
+        for c in self.callbacks:
+            try:
+                getattr(c, function_name)(**kwargs)
+            except Exception as e:
+                warnings.warn(
+                    f"Exception occurred during {function_name} with callback {type(c).__name__}, error message: {e}"
```

Comment:
Something like this ismore compact and informative.
```suggestion
                    f"Callback {type(c).__name__}.{function_name} raised {type(e).__name__}: {e}"
```

Reply:
My error message is more informative.